src/pages/batch.mdx

```diff
@@ -0,0 +1,136 @@
+
+## Limitations of Batches
+
+Jobs are designed to be long running HPC workloads and can take some time to spin up. They are not designed with reactivity in mind and are not suitable for responding to events from cloud resources.
+
+Jobs are unable to run the following:
+
+- Topic Subscriptions
+- Bucket Notifications
+- API & HTTP resources
+- Websocket message handlers
+
+Jobs can be used to read and write to/from all nitric resources, for example they can publish new messages to a Topic, read and write to a Bucket, or read and write to a Database. They just can't respond to real-time events from these resources.
+
+## Defining Batches
+
+Batches are defined similarly to services in a project's `nitric.yaml` file. For example:
+
+```yaml
+batches:
+  - match: ./batches/*.ts
+    start: yarn dev:batches $BATCH_PATH
+```
+
+<Note>Batches can contain any number of Job Definitions.</Note>
+
+## Defining a Job
+
+Within a Batch we create Job Definitions, by creating a new Job with a unique name and defining a handler function that will be executed when the job is submitted.
+
+<CodeGroup>
+
+```typescript
+import { job, JobContext } from '@nitric/sdk'
+
+const myJob = job('analyse')
+
+// Use `handler` to register the callback function that will run when a job is submitted
+myJob.handler(
+  async (ctx: JobContext) => {
+    // Do some work
+  },
+  { cpus: 1, memory: 1024, gpus: 0 }
+)
+```
+
+```python
+from nitric.resources import job
+from nitric.application import Nitric
+from nitric.context import JobContext
+
+analyze = job("analyze")
+
+# Create the callback function that will run when a job is submitted
+@analyze(cpus=1, memory=1024, gpus=0)
+async def generate_image(ctx: None):
+    # Do some work
+
+
+Nitric.run()
+```
+
+</CodeGroup>
+
+## Submitting Jobs for Execution
+
+Jobs may be submitted from Nitric `services` or other `batches` using the `submit` method on the job reference. When submitting a job you can provide a payload that will be passed to the job handler function.
+
+<Note>
+  When submitting a job, you can optionally override the default resources
+  allocated to the job.
+</Note>
+
+<CodeGroup>
+
+```javascript
+import * as nitric from '@nitric/sdk'
+
+const api = nitric.api('public')
+const analyze = nitric.job('analyze').allow('submit')
+
+api.post('/submit-job', async (ctx) => {
+  await analyze.submit(
+    {
+      someKey: 'someValue',
+    },
+    // optional job resource overrides
+    { cpus: 1, memory: 1024, gpus: 0 }
+  )
+})
+```
+
+```python
+from nitric.resources import api, job
+from nitric.application import Nitric
+
+analyze = job("analyze").allow("submit")
+public_api = api("public")
+
+@public_api.post("/submit-job")
+async def submit_job(ctx):
+    await analyze.submit(
+      {
+        "someKey": "someValue"
+      },
+      # optional job resource overrides
+      { cpus: 1, memory: 1024, gpus: 0 }
```

Comment:
considering removing the job resource overrides for now.

Have updated to implementation of the Job server in core to allow it to be persistent as well as one-shot (driven by the gateway). So in some cases depending on the implementation these runtime overrides may not be available/ignored.